src/firefly/js/visualize/saga/ChartsSync.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+import {take} from 'redux-saga/effects';
+
+import {has} from 'lodash';
+
+import {flux} from '../../Firefly.js';
+import * as TablesCntlr from '../../tables/TablesCntlr.js';
+import * as TableStatsCntlr from '../TableStatsCntlr.js';
+import * as TableUtil from '../../tables/TableUtil.js';
+
+import * as XYPlotCntlr from '../XYPlotCntlr.js';
+import * as HistogramCntlr from '../HistogramCntlr.js';
+
+
+
+/**
+ * this saga handles chart related side effects
+ */
+export function* syncCharts() {
+    var tableStatsState, xyPlotState, histogramState;
+
+    while (true) {
+        const action= yield take([TableStatsCntlr.SETUP_TBL_TRACKING, TablesCntlr.TABLE_NEW_LOADED]);
+        const request= action.payload.request;
+        switch (action.type) {
+            case TableStatsCntlr.SETUP_TBL_TRACKING:
+                const {tblId} = action.payload;
+                if (TableUtil.isFullyLoaded(tblId)) {
+                    TableStatsCntlr.dispatchLoadTblStats(TableUtil.findTblById(tblId).request);
+                }
+                break;
+            case TablesCntlr.TABLE_NEW_LOADED:
+                const {tbl_id} = action.payload;
+                tableStatsState = flux.getState()[TableStatsCntlr.TBLSTATS_DATA_KEY];
+                if (has(tableStatsState, tbl_id)) {
+                    TableStatsCntlr.dispatchLoadTblStats(request);
+                }
+                xyPlotState = flux.getState()[XYPlotCntlr.XYPLOT_DATA_KEY];
+                if (has(xyPlotState, tbl_id)) {
+                    const xyPlotParams = xyPlotState[tbl_id].xyPlotParams;
+                    XYPlotCntlr.dispatchLoadPlotData(xyPlotParams, request);
+                }
+                histogramState = flux.getState()[HistogramCntlr.HISTOGRAM_DATA_KEY];
+                if (has(histogramState, tbl_id)) {
+                    const histogramParams = xyPlotState[tbl_id].histogramParams;
+                    HistogramCntlr.dispatchLoadColData(histogramParams, request);
+                }
+                break;
+        }
+    }
+}
```

Comment:
Then we need an action for setting/destroying chart area. Or we have one?

2016-05-07 10:08 GMT-07:00 Trey Roby <notifications@github.com>:

> In src/firefly/js/visualize/saga/ChartsSync.js
> <https://github.com/Caltech-IPAC/firefly/pull/79#discussion_r62419585>:
>
> > +                    TableStatsCntlr.dispatchLoadTblStats(request);
> > +                }
> > +                xyPlotState = flux.getState()[XYPlotCntlr.XYPLOT_DATA_KEY];
> > +                if (has(xyPlotState, tbl_id)) {
> > +                    const xyPlotParams = xyPlotState[tbl_id].xyPlotParams;
> > +                    XYPlotCntlr.dispatchLoadPlotData(xyPlotParams, request);
> > +                }
> > +                histogramState = flux.getState()[HistogramCntlr.HISTOGRAM_DATA_KEY];
> > +                if (has(histogramState, tbl_id)) {
> > +                    const histogramParams = xyPlotState[tbl_id].histogramParams;
> > +                    HistogramCntlr.dispatchLoadColData(histogramParams, request);
> > +                }
> > +                break;
> > +        }
> > +    }
> > +}
>
> if you look at CatalogWatcher.js, you see the first line is:
>
> yield take(ImagePlotCntlr.PLOT_IMAGE);
>
> This keeps the sega from doing anything until there is a image loaded. You
> might be able to so something like that.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Caltech-IPAC/firefly/pull/79/files/d21b15a078bcd06a2bc7634b6e1a98769057da2e#r62419585>
>